kpatch-build/kpatch-elf.c

```diff
@@ -238,6 +239,16 @@ void kpatch_create_section_list(struct kpatch_elf *kelf)
 
 		if (!gelf_getshdr(scn, &sec->sh))
 			ERROR("gelf_getshdr");
+		else
+			sec->uncompressed_size = sec->sh.sh_size;
+
+		if ((sec->sh.sh_flags & SHF_COMPRESSED)) {
+			if (!gelf_getchdr(scn, &ch))
+				ERROR("gelf_getchdr");
+			else
```

Comment:
These else conditions aren't necessary because ERROR never returns.

Reply:
> These else conditions aren't necessary because ERROR never returns.

yet, the else condition had been remove.